include/flatbuffers/grpc.h

```diff
@@ -22,49 +22,198 @@
 #include "grpc++/support/byte_buffer.h"
 #include "grpc/byte_buffer_reader.h"
 
+namespace flatbuffers {
+namespace grpc {
+
+enum AddRef { ADD_REF };
+enum StealRef { STEAL_REF };
+
+template <class T>
+class Message {
+ public:
+  inline Message()
+      : slice_(grpc_empty_slice()) {
+  }
+
+  inline Message(grpc_slice slice, AddRef)
+      : slice_(grpc_slice_ref(slice)) {
+  }
+
+  inline Message(grpc_slice slice, StealRef)
+      : slice_(slice) {
+  }
+
+  inline Message(const Message &other)
+      : slice_(grpc_slice_ref(other.slice_)) {
+  }
+
+  inline Message(Message &&other)
+      : slice_(other.slice_) {
+    other.slice_ = grpc_empty_slice();
+  }
+
+  inline Message &operator=(const Message &other) {
+    slice_ = grpc_slice_ref(other.slice_);
+  }
+
+  inline Message &operator=(Message &&other) {
+    slice_ = other.slice_;
+    other.slice_ = grpc_empty_slice();
+  }
+
+  inline ~Message() {
+    grpc_slice_unref(slice_);
+  }
+
+  inline const grpc_slice& BorrowSlice() const {
+    return slice_;
+  }
+
+  inline bool Verify() const {
+    if (slice_.refcount != nullptr) {
+      Verifier verifier(slice_.data.refcounted.bytes, slice_.data.refcounted.length);
+      return verifier.VerifyBuffer<T>(nullptr);
+    } else {
+      Verifier verifier(slice_.data.inlined.bytes, slice_.data.inlined.length);
+      return verifier.VerifyBuffer<T>(nullptr);
+    }
+  }
+
+  inline const T* GetRoot() const {
+    if (slice_.refcount != nullptr) {
+      return flatbuffers::GetRoot<T>(slice_.data.refcounted.bytes);
+    } else {
+      return flatbuffers::GetRoot<T>(slice_.data.inlined.bytes);
+    }
+  }
+
+ protected:
+  grpc_slice slice_;
+};
+
+// This isnt a true allocator, as it only allows up to two oustanding
+// allocation at a time. However, this is sufficient for our use case, and
+// simplifies the grpc_slice tracking.
+class SliceAllocator : public simple_allocator {
+ public:
+  inline SliceAllocator()
+      : slices_{grpc_empty_slice(), grpc_empty_slice()} {
```

Comment:
Ah yes, good point. I'm realizing we don't have the GRPC code as part of our CI.. which is bad.. but yes, for the moment,.. we're cool with whatever C++11 GRPC uses :)